PsfLauncher/ErrorInformation.h

```diff
@@ -0,0 +1,85 @@
+#pragma once
+#include <string>
+#include <windows.h>
+#include <iostream>
+#include "utilities.h"
+
+class ErrorInformation
+{
+public:
+    ErrorInformation() : customMessage(L""), errorMessage(L""), isThereAnError(false)
+    {
+    }
+
+
+    ErrorInformation(std::wstring customMessage, DWORD errorNumber) :
+        ErrorInformation(customMessage, errorNumber, L"")
+    {
+    }
+
+    ErrorInformation(std::wstring customMessage, DWORD errorNumber, std::wstring exeName)
+        : customMessage(customMessage)
+        , errorNumber(errorNumber)
+        , errorMessage(widen(std::system_category().message(errorNumber)))
+        , isThereAnError(true)
+        , exeName(exeName)
+    {
+    }
+
+    ErrorInformation(const ErrorInformation &toCopy)
+    {
+        this->customMessage = toCopy.customMessage;
+        this->errorMessage = toCopy.errorMessage;
+        this->isThereAnError = toCopy.isThereAnError;
+        this->errorNumber = toCopy.errorNumber;
+    }
+
+    ErrorInformation& operator=(const ErrorInformation &toAssign)
+    {
+        if (this == &toAssign)
+        {
+            return *this;
+        }
+
+        this->customMessage.clear();
+        this->customMessage = toAssign.customMessage;
+        this->errorMessage = toAssign.errorMessage;
+        this->isThereAnError = toAssign.isThereAnError;
+        this->errorNumber = toAssign.errorNumber;
+
+        return *this;
+    }

```

Comment:
Neither of these copy over `exeName` - is this intentional? If it's not intentional, then neither of these functions are necessary since the compiler will generate the equivalent for you

Reply:
Though `exeName` doesn't seem to be used for anything? Could also remove it, in which case neither of these functions are necessary

---
In reply to: [285251717](https://github.com/microsoft/MSIX-PackageSupportFramework/pull/42#discussion_r285251717) [](ancestors = 285251717)

Reply:
Thank you for pointing that out.  I added exeName.